### DISCUSSION TITLE
Building for Windows x64 - Cannot generate CMake caches at CMakeLists.txt:34

### DISCUSSION THREAD

**Discussion by mhopper2** [SoftEtherVPN/SoftEtherVPN#1779] [Q&A]
I am following the steps outlined here:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_WINDOWS.md
It may be important to note I am working from Visual Studio 2022 and the options described in step 4 don't seem to exist.

Error description:
"Severity Code Description Project File Line Suppression State
Error CMake Error at src/Mayaqua/CMakeLists.txt:34 (message):
Required EVP_PKEY_get_raw_public_key() not found in OpenSSL library! C:\Users\12405\Source\Repos\SoftEtherVPN\src/Mayaqua/CMakeLists.txt 34"

This error happens in CMakeList.txt:34 in the following statement:
if(NOT HAVE_EVP_PKEY_GET_RAW_PUBLIC_KEY)
message(FATAL_ERROR "Required EVP_PKEY_get_raw_public_key() not found in OpenSSL library!")
endif()

What could be causing this error?

**Comment by chipitsine** [comment]
that error means we need OpenSSL-1.1.1 or higher built with EVP_PKEY enable (it is enabled by default in those libraries).

openssl is set using vcpkg. I'm not sure how good is documentation on vcpkg interoperation. feel free to report improvement there.

**Reply by mhopper2** [reply]
Are you saying this is an error during the vcpkg clone/integration step? 
How can I fix this error?

**Reply by shipitsin-ilia** [reply]
I tried on very fresh laptop.
I followed official Windows docs and I was able to build SE VPN using Visual Studio 2022.

did you do `vcpkg integrate install` step ? it should be invoked from elevated (i.e. admin) cmd prompt.
